asserts/confdb.go

```diff
@@ -167,6 +184,67 @@ func (cc *ConfdbControl) Serial() string {
 	return cc.HeaderString("serial")
 }
 
+// ConfdbControlGroup holds a single group in a confdb-control assertion.
+type ConfdbControlGroup struct {
+	Operators       []string
+	Authentications []string
+	Views           []string
+}
+
+// Groups returns the groups in the raw assertion's format.
```

Comment:
is this meant to be used on a client for printing?

Reply:
The [method](https://github.com/st3v3nmw/snapd/blob/11b14dd1d31673232043fcbfcfaf933ca10162b0/overlord/devicestate/devicemgr.go#L1968) to sign the `confdb-control` assertion takes `groups []interface{}` but we need `groups` to be in the SD172/raw format so that it can go through `asserts.assembleConfdbControl`. The`Groups` method converts the internal representation to the output representation. Its usage in the API would look something like [this](https://github.com/st3v3nmw/snapd/blob/11b14dd1d31673232043fcbfcfaf933ca10162b0/daemon/api_confdb.go#L194)

Apart from the device manager methods to sign the assertion, the rest is unmerged code I wrote for the demo

Reply:
your code is mutating the assertion? that is not something we can implement you need some intermediate object that can be mutated and then turn that back into an assertions. If an assertion is mutable in obvious ways bugs could make it mean things that were not actually signed